### DISCUSSION TITLE
Just curious on how close we are to the Rust release of Fish?

### DISCUSSION THREAD

**Discussion by justinfarrelldev** [fish-shell/fish-shell#10314] [General]
Hello,
I noticed the last release was early January and was just wondering when to expect the Rust release of Fish. I (and I'm sure many others) am super excited!

Thank you!

**Comment by faho** [comment]
In general the answer is "when it's done", we have learned not do do release estimates. In this particular case I don't believe it is particularly close.

We have "only" ported all the code, that's the first 90% of the work. Now we need the second 90%.

There are difficult questions like how to make it build and packageable (cargo unfortunately won't cut it - it does not have handling of non-binary assets, while we have ~1200 fish scripts plus documentation plus webconfig plus themes), there are a load of bugs left to shake out.

And, most importantly, there isn't currently really a convincing reason for any *user* to want to use it. It's buggier than 3.7.0 and doesn't really have any compelling new features (because those were backported to 3.7.0). So why release it when there's nothing *to* release?
